oak_runtime/src/node/http/mod.rs

```diff
@@ -439,11 +438,21 @@ impl Pipe {
 fn get_oak_label(req: &HttpRequest) -> Result<Label, OakStatus> {
     let headers = (
         req.headers
-            .get(oak_abi::OAK_LABEL_HTTP_JSON_KEY)
-            .map(|m| m.values.as_slice()),
+            .as_ref()
+            .map(|map| {
+                map.headers
+                    .get(oak_abi::OAK_LABEL_HTTP_JSON_KEY)
+                    .map(|m| m.values.as_slice())
+            })
+            .flatten(),
```

Comment:
Would it work to use `filter_map` on line 442, and avoid this flattening here?

Reply:
`Option` does not have a `filter_map`. (my earlier answer to the question about `#![feature(result_flattening)]` was misleading. Please see my updated response :))

Reply:
Sorry I meant `flat_map`, and anyways I thought it was an Iter. For Option, `and_then` should work, and is more idiomatic.

Reply:
Thanks.`and_then` works.